### DISCUSSION TITLE
Act as an intermediary for a BDR91

### DISCUSSION THREAD

**Discussion by roflson** [ghoti57/evofw3#40] [Q&A]
Hi, I'd like to know if it's conceptually possible to act as a man in the middle for a BDR91 (my heating zone valve/boiler control).

I've ordered one of the  SSM-D2 from your website and I'm hoping I can add a few features that the normal evohome controller doesn't have, but I'd like to keep my current BDR91s wired up how they are and somehow have the normal controller pair to me, and me pair to the BDR91.

Alternatively if that's not possible, can I replace the controller completely? Obviously I'd lose web access unless I implemented something myself but it will be worth it for the savings I should be able to generate from this. I think I could save £1000+ a year since one of my zones keeps my boiler running all day throughout the winter. 

Thanks!

**Comment by ghoti57** [comment]
I'm sure man in the middle is possible but you won't be able to implement anything on the SSM-D2.

evofw3 on that device has some quite time critical sections and the processor is really quite limited.  It doesn't understand anything about the meaning of the messages it handles, only the format of packets transmitted.  The processor doesn't have the guts to do anything more than RX and TX messages as a gateway device for whatever host system it's plugged into.

Any logic to achieve something useful would have to be in the host code.

I think related ideas have been been discussed in forums linked to the ramses_cc custom component for Home Assistant.
Have a look at the wiki for that code <https://github.com/zxdavb/ramses_cc> and repeat your question to its dev who can probably point you to people trying similar/related things.

**Reply by roflson** [reply]
Ah yes sorry I'm only using the SSM-D2 for the wireless RX/TX, I will be doing all of the actual logic on the host server. Thanks for the link, will take a look over there.
